cli/azd/cmd/build.go

```diff
@@ -166,15 +166,14 @@ func (ba *buildAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 			continue
 		}
 
-		buildTask := ba.serviceManager.Build(ctx, svc, nil)
-		go func() {
-			for buildProgress := range buildTask.Progress() {
-				progressMessage := fmt.Sprintf("Building service %s (%s)", svc.Name, buildProgress.Message)
-				ba.console.ShowSpinner(ctx, progressMessage, input.Step)
-			}
-		}()
-
-		buildResult, err := buildTask.Await()
+		var err error
+		ba.serviceManager.SetProgressDisplay(func(msg string) {
+			ba.console.ShowSpinner(
+				ctx,
+				fmt.Sprintf("Building service %s (%s)", svc.Name, msg),
+				input.Step)
+		})
+		buildResult, err := ba.serviceManager.Build(ctx, svc, nil)
```

Comment:
The `ServiceManager` is a business/backend component and should not be participating in any display related functionality.

Reply:
That is certainly true if we were thinking about moving our business logic into some other application and reusing it in those scenarios. Short of that, what do we gain by largely abstracting our layers?